nsxt/resource_nsxt_policy_service_test.go

```diff
@@ -575,6 +575,118 @@ func TestAccResourceNsxtPolicyService_algType(t *testing.T) {
 	})
 }
 
+func TestAccResourceNsxtPolicyService_nestedServiceType(t *testing.T) {
+	name := getAccTestResourceName()
+	updateName := getAccTestResourceName()
+	testResourceName := "nsxt_policy_service.test"
+	testNestedService1Name := "HTTP"
+	testNestedService2Name := "HTTPS"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtPolicyServiceCheckDestroy(state, updateName)
+		},
+		Steps: []resource.TestStep{
+			{
+				// Step 0: Create a nested service
+				Config: testAccNsxtPolicyNestedServiceCreateTemplate(name, testNestedService1Name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccNsxtPolicyServiceExists(testResourceName),
+					resource.TestCheckResourceAttr(testResourceName, "display_name", name),
+					resource.TestCheckResourceAttr(testResourceName, "description", "Nested service"),
+					resource.TestCheckResourceAttrSet(testResourceName, "path"),
+					resource.TestCheckResourceAttrSet(testResourceName, "revision"),
+					resource.TestCheckResourceAttr(testResourceName, "tag.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "algorithm_entry.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "l4_port_set_entry.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "icmp_entry.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "igmp_entry.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "ether_type_entry.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "ip_protocol_entry.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "nested_service_entry.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "nested_service_entry.0.display_name", testNestedService1Name),
+					resource.TestCheckResourceAttr(testResourceName, "nested_service_entry.0.description", "Entry-1"),
+					resource.TestCheckResourceAttr(testResourceName, "nested_service_entry.0.nested_service_path", "/infra/services/"+testNestedService1Name),
```

Comment:
unfortunately this would fail for global manager appliance, as `infra` is replaced with `global_infra` there. I'd suggest to use `TestMatchResourceAttr` or `TestCheckResourceAttrWith` to only match the service name without the path prefix.